python/mlcroissant/mlcroissant/_src/operation_graph/base_operation.py

```diff
@@ -9,56 +9,28 @@
 import networkx as nx
 
 from mlcroissant._src.structure_graph.base_node import Node
-from mlcroissant._src.structure_graph.nodes.record_set import RecordSet
 
 
 class Operations(nx.DiGraph):
-    """Overwrites nx.DiGraph to keep track of last operations."""
-
-    def last_operations(self, node: Node, only_leaf=False) -> list[Operation]:
-        """Retrieves the last operations for a node in the graph of operations.
-
-        Args:
-            node: The node of the operations.
-            only_leaf: Whether to only pick leaves operations (only_leaf=True) or to
-                visit upstream operations (only_leaf=False).
-        """
-        leaves = [operation for operation in self.nodes if self.is_leaf(operation)]
-
-        def is_ancestor(node1: Node, node2: Node) -> bool:
-            # node1 is predecessor of node2 iff node2 is in the descendants of node1.
-            ancestors = nx.ancestors(node2.ctx.graph, node2)
-            if isinstance(node1, RecordSet):
-                # If node1 is a RecordSet, we have to inspect its fields.
-                return any(is_ancestor(field1, node2) for field1 in node1.fields)
-            return node1 in ancestors
-
-        entry = self.entry_operations()[0]
-        operations: set[Operation] = set()
-        for leaf in leaves:
-            if is_ancestor(leaf.node, node):
-                operations.add(leaf)
-            elif not only_leaf:
-                # We explore upstream operations until we find the first operation that
-                # matches the node.
-                try:
-                    for operation in reversed(
-                        list(nx.shortest_path(self, entry, leaf))
-                    ):
-                        if is_ancestor(operation.node, node):
-                            operations.add(operation)
-                            break
-                except nx.exception.NetworkXNoPath:
-                    continue
-        return list(operations)
+    """Overwrites nx.DiGraph to keep track of last operations.
+
```

Comment:
operations?

Reply:
Or "pointer to the chain of last operations"?

Reply:
Done.